### DISCUSSION TITLE
XML Object Length Returning 0

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5883] [Q&A] [answered]
I have the below code in the response transformer for an API call that returns JSON. The JSON contains encoded CCDs, so I am trying to decode them and they get the component count for each CCD. For some reason though my code is always returning 0 for the componentCount channelMap. Does anyone know why that would be happening?

    for(i=0; i<msg.length; i++){
	    var clinicalDocumentDecoded = new java.lang.String(FileUtil.decode(msg[i]['data'].toString()), 'utf-8')
	    var clinicalDocumentXML = new XML(clinicalDocumentDecoded)
	
	    channelMap.put('document' + i, clinicalDocumentXML)
	    channelMap.put('type' + i, typeof(clinicalDocumentXML))
	    channelMap.put('componentCount' + i, clinicalDocumentXML['component'].length())
    }

**Comment by pacmano1** [comment, marked as answer]
Probably you need to set the XML namespace.

Not your code but:

```JS
for (i = 0; i < msg.length; i++) {
    var decodedString = new java.lang.String(FileUtil.decode(msg[i]['base64_ccda'].toString()), 'utf-8')
    $c('cda' + i, decodedString);
    $c('cdaXML' + i, new XML(decodedString))
    default xml namespace = new Namespace('urn:hl7-org:v3')
    $c('componentCount' + i, $c('cdaXML' + i)['component']['structuredBody']['component'].length())
}
```

**Reply by JJWest12** [reply]
Without even trying it that is 100% the issue. I ALWAYS forget that part lol. Thank you!
